src/aiidalab_qe/workflows/__init__.py

```diff
@@ -141,10 +141,22 @@ def get_builder_from_protocol(
                     hubbard_type="U",
                     use_kinds=True,
                 )
-            hubbard_structure.store()
-            builder.structure = hubbard_structure
+            if hubbard_structure.hubbard == structure.hubbard:
+                builder.structure = structure
+            else:
+                hubbard_structure.store()
+                builder.structure = hubbard_structure
```

Comment:
Raise error when `structure` is a normal StructureData, which doesn't have `hubbard`. I think you don't need check the `hubbard` here, because you need use the newly created `hubbard_structure`.
```suggestion
            hubbard_structure.store()
            builder.structure = hubbard_structure
```